scripts/results/plot_maxfilter.py

```diff
@@ -11,65 +12,94 @@
 from mne import Epochs
 from mne.preprocessing import maxwell_filter
 
+import matplotlib.pyplot as plt
+import numpy as np
+
 from library.config import study_path, cal, ctc, set_matplotlib_defaults
 
 event_ids = [5, 6, 7]  # Famous faces
 
 subject = "sub001"
 bads = ['MEG1031', 'MEG1111', 'MEG2113']
-filter_params = dict(fir_window='hamming', l_trans_bandwidth=0.5, phase='zero',
-                     h_trans_bandwidth='auto', filter_length='auto',
+filter_params = dict(fir_window='hamming', filter_length='auto', phase='zero',
+                     l_trans_bandwidth='auto', h_trans_bandwidth='auto',
                      fir_design='firwin')
 
 raw_fname_in = op.join(study_path, 'ds117', subject, 'MEG', 'run_01_raw.fif')
-sss_fname_in = op.join(study_path, 'ds117', subject, 'MEG', 'run_01_sss.fif')
+sss_fname_in = op.join(study_path, 'ds117', subject, 'MEG', 'run_%02d_sss.fif')
 
 ###############################################################################
 # First the filtered raw data.
 raw = mne.io.read_raw_fif(raw_fname_in, preload=True)
 
 raw.info['bads'] = bads
-picks = mne.pick_types(raw.info, meg=True, exclude='bads')
 raw.filter(1., 40, **filter_params)
 
 events = mne.find_events(raw, stim_channel='STI101', consecutive='increasing',
                          mask=4352, mask_type='not_and', min_duration=0.003,
                          verbose=True)
-evoked_before = Epochs(raw, events, event_id=event_ids, picks=picks).average()
+evoked_before = Epochs(raw, events, event_id=event_ids).average()
 
 ###############################################################################
 # Then Maxfiltered and SSS'd data.
+
 raw = mne.io.read_raw_fif(raw_fname_in, preload=True)
-raw_sss = mne.io.read_raw_fif(sss_fname_in, preload=True)
+raw_sss_mf = mne.io.read_raw_fif(sss_fname_in % 1, preload=True,
+                                 verbose='error')
 raw.info['bads'] = bads
-raw_sss.info['bads'] = bads
+raw.fix_mag_coil_types()
+# Get the origin and destinations they used
+destination = raw_sss_mf.info['dev_head_t']
+origin = raw_sss_mf.info['proc_history'][0]['max_info']['sss_info']['origin']
+# Get the head positions.
+# Usually this can be done by saving to a text file with MaxFilter
+# instead, and reading with :func:`mne.chpi.read_head_pos`.
+chpi_picks = mne.pick_types(raw_sss_mf.info, meg=False, chpi=True)
+assert len(chpi_picks) == 9
+head_pos, t = raw_sss_mf[chpi_picks]
+# Add first_samp.
+t = t + raw_sss_mf.first_samp / raw_sss_mf.info['sfreq']
+# The head positions in the FIF file are all zero for invalid positions
+# so let's remove them, and then concatenate our times.
+mask = (head_pos != 0).any(axis=0)
+head_pos = np.concatenate((t[np.newaxis], head_pos)).T[mask]
+# In this dataset due to old MaxFilter (2.2.10), data are uniformly
+# sampled at 1 Hz, so we save some processing time in
+# maxwell_filter by downsampling.
+skip = int(round(raw_sss_mf.info['sfreq']))
+head_pos = head_pos[::skip]
```

Comment:
Just wondering ... this couldn't be a function in MNE?

Reply:
It could, but most users shouldn't need to do it. The right way (also in the comments) is to use `maxfilter -hp` and `read_head_pos`, but here it's inconvenient because we have the matching / incompleteness problem of the FTP site files, and even if we got them working we'd have to distribute them here.

Reply:
Well, we could have fetched it using `urllib` but anyway doesn't the strategy your propose mean that the user would need to have an Elekta maxfilter license?

Reply:
All strategies we currently support (the one used here and the one in comments here) for doing movement compensation currently require a MaxFilter license, yes

We have our own position estimation code but it's private and not extensively tested enough for public consumption

Reply:
Oh, so the estimated head positions are saved in the CHPI channels? makes sense. I guess that would be a caveat to add then to the manuscript. It's comparable up to the estimation of head position